### PR TITLE
Add support for NOT NULL in DDL statements

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

**Comment by electrum** [comment]
@findepi I already reviewed this in the other repo. Let me know if you'd like to take a look, otherwise I'll merge this.

File: presto-parser/src/main/java/io/prestosql/sql/testing/TreeAssertions.java [hunk 1]
```diff
@@ -63,9 +63,8 @@ private static Statement parseFormatted(SqlParser sqlParser, ParsingOptions pars
             return sqlParser.createStatement(sql, parsingOptions);
```
Comment (kokosing) [inline presto-parser/src/main/java/io/prestosql/sql/testing/TreeAssertions.java]: Can you please add to commit messages of all commits:

```
Extracted from: https://github.com/prestodb/presto/pull/11561
```

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java [hunk 2]
```diff
@@ -209,7 +210,8 @@ public void destroy()
                     // skip unsupported column types
                     if (columnMapping.isPresent()) {
                         String columnName = resultSet.getString("COLUMN_NAME");
-                        columns.add(new JdbcColumnHandle(columnName, typeHandle, columnMapping.get().getType()));
+                        boolean nullable = (resultSet.getInt("NULLABLE") == columnNullable);
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java]: or `== columnNullableUnknown`

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 3]
```diff
@@ -187,4 +195,48 @@ else if (varcharType.getBoundedLength() <= 16777215) {
...
+            }
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    public void renameTable(JdbcIdentity identity, JdbcTableHandle handle, SchemaTableName newTableName)
+    {
+        // MySQL doesn't support specifying the catalog name in a rename. By setting the
+        // catalogName parameter to null, it will be omitted in the ALTER TABLE statement.
```
Comment (findepi) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java]: we should `checkArgument(handle.getSchemaName() eq newTableName.getSchema())`

File: presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestMySqlTypeMapping.java [hunk 4]
```diff
@@ -117,7 +117,7 @@ public void testMySqlCreatedParameterizedVarchar()
                 .addRoundTrip(stringDataType("mediumtext", createVarcharType(16777215)), "c")
                 .addRoundTrip(stringDataType("longtext", createUnboundedVarcharType()), "d")
                 .addRoundTrip(varcharDataType(32), "e")
-                .addRoundTrip(varcharDataType(20000), "f")
+                .addRoundTrip(varcharDataType(15000), "f")
```
Comment (findepi) [inline presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestMySqlTypeMapping.java]: Is this related to `Update testing-mysql-server to 8.0.12` ?

**Comment by findepi** [comment]
I don't like how the "Add DDL support for JDBC connectors" commit contains several changes that are not directly related.
No change requested.

**Comment by findepi** [comment]
@electrum i only skimmed. LGTM.

File: presto-parser/src/main/java/io/prestosql/sql/testing/TreeAssertions.java (hunk 1 above)
Comment (electrum) [reply presto-parser/src/main/java/io/prestosql/sql/testing/TreeAssertions.java]: I don't think it's needed for a one line fix

**Comment by electrum** [comment]
> Have you changed anything other than related to rebase?

@kokosing The only change was I moved `testInsertIntoNotNullColumn` to be MySQL and PostgreSQL specific (and thus removed the `AbstractTestDistributedQueries` changes) as it was making assumptions about the database-specific error message (and in the process was not testing it very well)

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java (hunk 3 above)
Comment (electrum) [reply presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java]: Perhaps I'm not following, but they don't have to be the same, as it's legal to rename a table across databases/schemas.

File: presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestMySqlTypeMapping.java (hunk 4 above)
Comment (electrum) [reply presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestMySqlTypeMapping.java]: Yes, I tried the original and it fails with

> Column length too big for column 'col_5' (max = 16383); use BLOB or TEXT instead

**Pull request merged**